quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/ConfigArgsConfigSource.java

```diff
@@ -75,58 +98,87 @@ public String getValue(String propertyName) {
-        String args = Environment.getConfigArgs();
+        // init here because the class might be loaded by CL without init
+        IGNORED_ARGS = asList("--verbose", "-v", "--help", "-h", AUTO_BUILD_OPTION_LONG, AUTO_BUILD_OPTION_SHORT);
+        String rawArgs = getRawConfigArgs();
         
-        if (args == null || "".equals(args.trim())) {
+        if (rawArgs == null || "".equals(rawArgs.trim())) {
             log.trace("No command-line arguments provided");
             return Collections.emptyMap();
         }
-        
+
         Map<String, String> properties = new HashMap<>();
 
-        for (String arg : ARG_SPLIT.split(args)) {
+        parseRawCliArgs(new BiConsumer<String, String>() {
+            @Override
+            public void accept(String key, String value) {
+                key = NS_KEYCLOAK_PREFIX + key.substring(2);
+
+                log.tracef("Adding property [%s=%s] from command-line", key, value);
+                properties.put(key, value);
+
+                String mappedPropertyName = getMappedPropertyName(key);
+
+                properties.put(mappedPropertyName, value);
+
+                PropertyMapper mapper = PropertyMappers.getMapper(mappedPropertyName);
+
+                if (mapper != null) {
+                    properties.put(mapper.getFrom(), value);
+                }
+
+                // to make lookup easier, we normalize the key
+                properties.put(Picocli.normalizeKey(key), value);
+            }
+        });
+
+        return properties;
+    }
+
+    public static void parseRawCliArgs(BiConsumer<String, String> cliArgConsumer) {
+        String rawArgs = getRawConfigArgs();
+
+        if (rawArgs == null || "".equals(rawArgs.trim())) {
+            log.trace("No command-line arguments provided");
+            return;
+        }
+
+        String[] args = ARG_SPLIT.split(rawArgs);
+
+        for (int i = 0; i < args.length; i++) {
+            String arg = args[i];
+
+            if (IGNORED_ARGS.contains(arg)) {
+                continue;
+            }
+
             if (!arg.startsWith(ARG_PREFIX)) {
-                throw new IllegalArgumentException("Invalid argument format [" + arg + "], arguments must start with '--'");
+                continue;
             }
 
             String[] keyValue = ARG_KEY_VALUE_SPLIT.split(arg);
             String key = keyValue[0];
-            
+
             if ("".equals(key.trim())) {
                 throw new IllegalArgumentException("Invalid argument key");
             }
-            
+
             String value;
-            
+
             if (keyValue.length == 1) {
-                continue;
+                if (args.length <= i + 1) {
+                    continue;
+                }
+                value = args[i + 1];
             } else if (keyValue.length == 2) {
                 // the argument has a simple value. Eg.: key=pair
                 value = keyValue[1];
```

Comment:
fyi: this is not enough validation. e.g. `./kc.sh start-dev --db-password=` doesstill work. But it's a good place to check for it :) I have some changes in my branch which we should merge later on, which also do some sanity checking of the actual value.